src/magic_folder/tahoe_client.py

```diff
@@ -164,6 +164,42 @@ def create_mutable_directory(self):
         capability_string = yield _get_content_check_code({OK, CREATED}, response)
         returnValue(capability_string)
 
+
+    @inlineCallbacks
+    def add_entry_to_mutable_directory(self, mutable_cap, path_name, entry_cap):
+        """
+        Adds an entry to a mutable directory
+
+        :param bytes mutable_cap: the capability-string of a mutable
+            to add an entry into
+
+        :param unicode path_name: the name of the entry (i.e. the path
+            segment)
+
+        :param bytes entry_cap: the capability of the entry (could be
+            any sort of capability).
+
+        :return Deferred[None]: or exception on error
+        """
+        post_uri = self.url.replace(
+            path=(u"uri", mutable_cap.decode("utf8"), path_name),
+            query=[
+                (u"t", u"uri"),
+                (u"replace", u"false"),
+            ],
+        )
+        response = yield _request(
+            self.http_client,
+            b"PUT",
+            post_uri,
+            data=entry_cap,
+        )
+        # Response code should probably be CREATED but it seems to be OK
+        # instead.  Not sure if this is the real Tahoe-LAFS behavior or an
+        # artifact of the test double.
```

Comment:
Maybe worth splitting add_entry_to_mutable_directory off into a separate PR with testing/web.py changes and other test coverage for it.  I'm kind of confused how it works at all right now... testing/web.py certainly doesn't pay any particular attention to `replace=false`.  I think it probably doesn't actually do anything with directories.

Also compare and contrast with `magic_folder.cli.Node.add_file` and `magic_folder.cli.TahoeClient.add_file`...

Reply:
I was basically just trans-literating code from what existed before (I thought) -- maybe we can just use `TahoeClient.add_file` directly here instead?

Reply:
This says `replace=false` has meaning: https://github.com/tahoe-lafs/tahoe-lafs/blob/master/docs/frontends/webapi.rst#attaching-an-existing-file-or-directory-by-its-read--or-write-cap

(I think the point here is that we WANT it to fail if we're trying to overwrite an entry, because that means that we're trying to add a user with the same name..)

Reply:
Yes, it means something to Tahoe-LAFS but I specifically meant that it is meaningless to testing/web.py.  If we need to use it (and I agree we don't want to allow silent overwrites in the normal case), we should expand testing/web.py to recognize it and emulate the real Tahoe-LAFS behavior.

Reply:
I'm not sure what you're getting at with the two `.add_file` methods?